gwak/data/prior.py

```diff
@@ -82,6 +109,141 @@ def __init__(self):
             psi = Uniform(0, 2 * torch.pi)
         )
 
+class LAL_BBHPrior(BasePrior):
+    
+    def __init__(
+        self,
+        f_min=30,
+        f_max=2048,
+        duration=2, # duration of the time series
+        f_ref=20.0
+    ):
+
+        self.priors = {}
+        self.wave_params = {}
+        self.sampled_params = {}
+        
+        self.lal_keys = [
+            "incl", # Transformed inclination angle. (TensorType)
+            "s1x", # Spin component x of the first BH. (TensorType)
+            "s1y", # Spin component y of the first BH. (TensorType)
+            "s1z", # Spin component z of the first BH. (TensorType)
+            "s2x", # Spin component x of the second BH. (TensorType)
+            "s2y", # Spin component y of the second BH. (TensorType)
+            "s2z", # Spin component z of the second BH. (TensorType)
+        ]
+
+        # Frequency series in Hz. (TensorType)
+        self.sampled_params["fs"] = torch.arange(f_min, f_max, 1 / duration) 
+        
+        # Chirp mass in solar masses. (TensorType)
+        self.priors['chirp_mass'] = Uniform(15, 30) 
+        
+        # Mass ratio m1/m2. (TensorType)
+        self.priors['mass_ratio'] = Uniform(0.5, 0.99) 
+        
+        # Luminosity distance in Mpc.(TensorType)
+        self.priors["dist_mpc"] = Uniform(50, 200) 
+        
+        # Coalescence time. (TensorType)
+        self.priors["tc"] = Constant(0) 
+        
+        # Sky location: Right ascension angle 
+        self.priors['ra'] = Uniform(0, 2*np.pi)
+        
+        # Sky location: Declination angle
+        self.priors['dec'] = Cosine(-np.pi/2, np.pi/2)
+        
+        # Phase of the two polarlization
+        self.priors['psi'] = Uniform(0, 2*np.pi)
+        
+        # ----- Spin & incl parameters (Bilby parameters) -----
+        # Inclination in bibly setup
+        self.priors['theta_jn'] = Sine() 
+        
+        # Spin phase angle
+        self.priors['phi_jl'] = Uniform(0, 2 * np.pi) 
+        
+        # Primary object tilt
+        self.priors['tilt_1'] = Sine(0, np.pi) 
+        
+        # Secondary object tilt
+        self.priors['tilt_2'] = Sine(0, np.pi) 
+        
+        # Relative spin azimuthal angle
+        self.priors['phi_12'] = Uniform(0, 2 * np.pi) 
```

Comment:
@AndyC80297 can you please create an issue describing the work that needs to be done to fix the BBH spin implementation